### PR TITLE
Voting methods for feature ranking in efs

### PR DESCRIPTION
**Pull request by bblodfon**
- Use `fastVoteR`, where 4 voting theory methods are now implemented in `Rcpp`
- Add `embedded_ensemble_fselect()`
- Refactoring/Simplified code on both ensemble feature selection functions and `EnsembleFSResult()`

### CODE REVIEW

File: R/embedded_ensemble_fselect.R [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+#'   eefsr
+#' }
+embedded_ensemble_fselect = function(
+    task,
+    learners,
+    init_resampling,
+    measure,
+    store_benchmark_result = TRUE
+  ) {
+  # TOCHECK: DO WE NEED callbacks?
```
Comment (be-marc) [inline R/embedded_ensemble_fselect.R:62]: Why not?

Comment (bblodfon) [reply R/embedded_ensemble_fselect.R:62]: in `ensemble_fselect` we were constructing auto fs tuner objects that used these callbacks, I don't see where to use them here.

File: R/EnsembleFSResult.R [hunk 2]
```diff
@@ -62,26 +72,53 @@ EnsembleFSResult = R6Class("EnsembleFSResult",
...
-    initialize = function(result, features, benchmark_result = NULL, measure_id,
-                          minimize = TRUE) {
+    #' @param measure ([mlr3::Measure])\cr
+    #'  The measure used to score the learners on the test sets generated
+    #'  during the ensemble feature selection process.
+    #'  This will be the 'active' measure used in methods of this object, but this
+    #'  can be changed with `$set_active_measure()`.
+    #' @param inner_measure ([mlr3::Measure])\cr
+    #'  The inner measure used to optimize and score the learners on the train sets
+    #'  generated during the ensemble feature selection process.
```
Comment (be-marc) [inline R/EnsembleFSResult.R:94]: Can we say that differently? Scoring on a train set sounds wrong. Is this the outer train set which is split by the inner resampling? We score the inner resample result?

Comment (bblodfon) [reply R/EnsembleFSResult.R:94]: Yes, its the outer train set. The `inner_resampling` generates N train/test splits. The `inner_measure` is used to optimize/tune on the train set and you get the best subset and final model + score on that train set. We use these final models to also score the corresponding test splits (the inner resampling result you ask), with the `measure`. In embedded `efs` we only do the second (no `inner_measure` is needed/used).

Comment (bblodfon) [reply R/EnsembleFSResult.R:94]: can change the wording to specifically mention the train/test splits of the inner resampling (I also mentionthat earlier in the doc), what do you think?

Comment (be-marc) [reply R/EnsembleFSResult.R:94]: > you get the best subset and final model + score on that train set

It is the final model with the best subset and corresponding performance estimated on the inner resampling. There is no scoring on the outer training set but scoring on the inner resampling result. This is very similar to nested resampling. Maybe stick to the words used bellow figure 4.5

https://mlr3book.mlr-org.com/chapters/chapter4/hyperparameter_optimization.html#sec-nested-resampling

Comment (bblodfon) [reply R/EnsembleFSResult.R:94]: Yes sorry Marc, it's as you say, when I was writing the above comment, I meant outer resampling (what we call `init_resampling`) as the one that generates the train/test splits. And yes, pretty much we are doing nested CV, with outer resampling the N times holdout split. I will update the doc

**Pull request merged**